src/main/java/seedu/duke/Parser.java

```diff
@@ -95,10 +96,12 @@ public Command parse(String fullCommand) throws Exception {
             Constructor<?> constructor = cls.getDeclaredConstructor(Ui.class, Data.class, HashMap.class);
             Object obj = constructor.newInstance(ui, data, arguments);
             return (Command) obj;
+        } catch (ClassNotFoundException e) {
+            // *Command class cannot be found!
+            throw new InvalidInputException(InvalidInputException.Type.UNKNOWN_COMMAND, e);
         } catch (Exception e) {
-            // If any exception thrown above, it means the command is not formatted properly
-            // TODO: Exception handling using a custom exception
-            throw new Exception(Constants.EXCEPTION_PARSER_INVALIDCOMMAND, e);
+            // Some other weird error occurred here
+            throw new UnknownException(e);
```

Comment:
Since you have implemented a `UnknownException` to catch all possible errors, should the `PatientManager` class be updated to catch `UnknownException` instead of `Exception`?